Woof/Woof/ViewModel/SitterProfileViewModel.swift

```diff
@@ -38,34 +48,47 @@ final class SitterProfileViewModel: ObservableObject {
         }
     }
 
-    /// Requests model layer to save modified data.
-    func save() {
+    /// Requests model layer to upload and save modified data.
+    @MainActor func save() async {
+        isSavingData = true
+
         var newSitter = Sitter()
         newSitter.name = name
         newSitter.surname = surname
         newSitter.phone = phone
         newSitter.bio = bio
         newSitter.pricePerHour = Double(pricePerHour) ?? 0
 
-        guard let data = try? JSONEncoder().encode(newSitter) else {
-            return
+        do {
+            try await upload(newSitter)
+            try await saveLocally(newSitter)
+        } catch {
```

Comment:
It seems more consequent.
```suggestion
            try await saveLocally(newSitter)
            currentSitter = newSitter
            isEditingMode = false
        } catch {
```

Reply:
Fixed with 7999620